src/apply.jl

```diff
@@ -117,28 +67,44 @@ function full_update_bp(
     apply_kwargs...,
   )
   if symmetrize
-    Rᵥ₁, Rᵥ₂ = symmetric_factorize(
-      Rᵥ₁ * Rᵥ₂, inds(Rᵥ₁); tags=edge_tag(v⃗[1] => v⃗[2]), observer!, apply_kwargs...
+    Rᵥ₁, Rᵥ₂ = factorize_svd(
+      Rᵥ₁ * Rᵥ₂,
+      inds(Rᵥ₁);
+      ortho="none",
+      tags=edge_tag(v⃗[1] => v⃗[2]),
+      singular_values!,
+      apply_kwargs...,
     )
   end
   ψᵥ₁ = Qᵥ₁ * Rᵥ₁
   ψᵥ₂ = Qᵥ₂ * Rᵥ₂
   return ψᵥ₁, ψᵥ₂
 end
 
-function simple_update_bp_full(o, ψ, v⃗; envs, (observer!)=nothing, apply_kwargs...)
+function simple_update_bp_full(o, ψ, v⃗; envs, (singular_values!)=nothing, apply_kwargs...)
   cutoff = 10 * eps(real(scalartype(ψ)))
-  regularization = 10 * eps(real(scalartype(ψ)))
   envs_v1 = filter(env -> hascommoninds(env, ψ[v⃗[1]]), envs)
   envs_v2 = filter(env -> hascommoninds(env, ψ[v⃗[2]]), envs)
-  sqrt_and_inv_sqrt_envs_v1 =
-    sqrt_and_inv_sqrt.(envs_v1; ishermitian=true, cutoff, regularization)
-  sqrt_and_inv_sqrt_envs_v2 =
-    sqrt_and_inv_sqrt.(envs_v2; ishermitian=true, cutoff, regularization)
-  sqrt_envs_v1 = first.(sqrt_and_inv_sqrt_envs_v1)
-  inv_sqrt_envs_v1 = last.(sqrt_and_inv_sqrt_envs_v1)
-  sqrt_envs_v2 = first.(sqrt_and_inv_sqrt_envs_v2)
-  inv_sqrt_envs_v2 = last.(sqrt_and_inv_sqrt_envs_v2)
+  sqrt_envs_v1 = [
+    ITensorsExtensions.map_eigvals(
+      sqrt, env, first(inds(env)), last(inds(env)); cutoff, ishermitian=true
```

Comment:
Instead of `first(inds(env)), last(inds(env))` I think it is better to have a check that `ndims(env) == 2` and then you can just use calls like `ITensorsExtensions.map_eigvals(sqrt, env, inds(env)...; ...)`.